sklearn/linear_model/ridge.py

```diff
@@ -1324,17 +1326,21 @@ def _solve_eigen_covariance_intercept(
         return (1 - hat_diag) / alpha, (y - y_hat) / alpha
 
     def _solve_eigen_covariance(
-            self, alpha, y, sqrt_sw, X_mean, s, V, X):
-        """Compute dual coefficients and diagonal of (Identity - Hat_matrix)
+            self, alpha, y, sqrt_sw, X_mean, eigvals, Q, X):
```

Comment:
here maybe keep the notation `V` instead of `Q`? since they are the right singular vectors of `X`